www/nginx/src/opnsense/mvc/app/models/OPNsense/Nginx/Nginx.xml

```diff
@@ -773,6 +773,27 @@
         <default>0</default>
         <Required>Y</Required>
       </https_only>
+      <ssl_protocols type="OptionField">
+        <multiple>Y</multiple>
+        <Sorted>Y</Sorted>
+        <OptionValues>
+          <TLSv1 value="TLSv1">TLSv1</TLSv1>
+          <TLSv1_1 value="TLSv1.1">TLSv1.1</TLSv1_1>
+          <TLSv1_2 value="TLSv1.2">TLSv1.2</TLSv1_2>
+          <TLSv1_3 value="TLSv1.3">TLSv1.3</TLSv1_3>
+        </OptionValues>
+        <Required>N</Required>
+        <default>TLSv1.2,TLSv1.3</default>
+      </ssl_protocols>
+      <ssl_ciphers type="TextField">
+        <default>ECDHE-ECDSA-CAMELLIA256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-CAMELLIA256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-ECDSA-CAMELLIA128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-RSA-CAMELLIA128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-ECDSA-CAMELLIA256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-CAMELLIA256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-ECDSA-CAMELLIA128-SHA256:ECDHE-RSA-AES128-SHA256</default>
+        <Required>N</Required>
+        <mask>/^(0x[0-9a-fA-F]{4,4}|[a-zA-Z_\-0-9]+)(?::((0x[0-9a-fA-F]{4,4}|[a-zA-Z_\-0-9]+)))*$/</mask>
```

Comment:
I am not sure here but maybe this regex implementation supports that you have an "i" at the end to match not case sensitive. This way you can simplify all groups and if you move the minus to the end, you won't have to escape it.

This is by the way heavily based on the mozilla generator:
https://ssl-config.mozilla.org/#server=nginx&version=1.17.7&config=modern&openssl=1.0.2&guideline=5.6

however the mozilla tool is likely currently outdated since it is based on nginx 1.17.1. I do have currently 1.20.x on my linux machine installed.

you will also never see anything prefixed with ssl_ here. That is on purpose. SSL is dead.
The UI does not need it, the xml config as well. So keep it only in the nginx.conf where it is needed.
If you think, you need a prefix, use tls_ instead.

Reply:
Maybe add the DH curves then as well.

Reply:
Hi!

> I am not sure here but maybe this regex implementation supports that you have an "i" at the end to match not case sensitive. This way you can simplify all groups and if you move the minus to the end, you won't have to escape it.

I just brazenly copied this regex from the tls_fingerprints section) I like this regex: 
`/^((((!|\+|-)?[A-Z][A-Z\d\+-]+)|(@STRENGTH)):?)*$/i`
can you please check this?

> Maybe add the DH curves then as well.

Certainly! I'll try

> you will also never see anything prefixed with ssl_ here. That is on purpose. SSL is dead.
The UI does not need it, the xml config as well.

used the ssl_ prefix for the sake of better reflecting the parameters in the nginx.conf only (it seemed that this would make it easier to understand it in future). will take it away 

> This is by the way heavily based on the mozilla generator

sorry, didn't quite understand this part. is this an introduction to the call not to use ssl_?)

Reply:
fir the last question: the default is from Mozilla. See the link.

Reply:
@fabianfrz 
replaced ssl with tsl in the model and form. regex replaced (works fine for me).
default ssl_ciphers are kept so as not to break existing configs